common/component/redis/v9client.go

```diff
@@ -20,13 +20,21 @@ import (
 	"time"
 
 	v9 "github.com/redis/go-redis/v9"
+
+	"github.com/dapr/kit/logger"
 )
 
 type v9Pipeliner struct {
 	pipeliner    v9.Pipeliner
 	writeTimeout Duration
 }
 
+var v9logger = logger.NewLogger("dapr.components.redisv9")
+
+const (
+	tokenRefreshInterval = 50 * time.Minute
```

Comment:
Apparently we do not have to do this every hour -- instead we should examine the most recent token expiration / duration, and then refresh the token say 10 minutes before it expires.

The token validity is variable and can be longer than one hour.

There should be a function you can use to examine the validity of the token.